pkg/bundle/internal/target/target.go

```diff
@@ -396,3 +417,48 @@ func (b *Data) Populate(pool *util.CertPool, formats *trustapi.AdditionalFormats
 	}
 	return nil
 }
+
+// Calculate hash from the given password string.
+func calculateHash(passwd string) string {
+	crcTable := crc32.MakeTable(128)
+	crc := crc32.New(crcTable)
+	crc.Write([]byte(passwd))
+
+	return strconv.Itoa(int(crc.Sum32()))
+}
+
+// Retrieve old password hashes from the given annotations
+// Annotation keys are taken from bundle additional formats keys
+func getOldHashAnnotations(annotations map[string]string, bundle *trustapi.AdditionalFormats) map[string]string {
+	var passwdHashAnnotations = make(map[string]string)
+
+	if bundle.JKS != nil {
+		passwdHashAnnotations[bundle.JKS.Key] = annotations[bundle.JKS.Key]
+	}
+	if bundle.PKCS12 != nil {
+		passwdHashAnnotations[bundle.PKCS12.Key] = annotations[bundle.PKCS12.Key]
+	}
+
+	return passwdHashAnnotations
+}
+
+// Calculate new password hash annotations from the given additional formats
+// keys and passwords
+func calcNewHashAnnotations(bundle *trustapi.AdditionalFormats) map[string]string {
+	var passwdHashAnnotations = make(map[string]string)
+
+	if bundle.JKS != nil && bundle.JKS.Password != nil {
+		passwdHashAnnotations[bundle.JKS.Key] = calculateHash(*bundle.JKS.Password)
```

Comment:
```suggestion
		passwdHashAnnotations["trust.cert-manager.io/jks-pwd-hash"] = calculateHash(*bundle.JKS.Password)
```

I don't think these keys should be dynamic. Please use well-known keys. I've added a suggestion, but feel free to suggest something else.